consensus/spos/debug/equivalentMessagesDebugger.go

```diff
@@ -2,67 +2,120 @@ package debug
 
 import (
 	"fmt"
+	"sync"
 
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data"
 	"github.com/multiversx/mx-chain-core-go/display"
-	"github.com/multiversx/mx-chain-go/consensus"
+	"github.com/multiversx/mx-chain-go/consensus/spos"
+	"github.com/multiversx/mx-chain-go/sharding"
 	logger "github.com/multiversx/mx-chain-logger-go"
 )
 
 var log = logger.GetOrCreate("debug/equivalentmessages")
 
+type proofsPoolHandler interface {
+	GetAllNotarizedProofs(shardID uint32) (map[string]data.HeaderProofHandler, error)
+	GetNotarizedProof(shardID uint32, headerHash []byte) (data.HeaderProofHandler, error)
+	IsInterfaceNil() bool
+}
+
 type equivalentMessagesDebugger struct {
+	proofsPool       proofsPoolHandler
+	shardCoordinator sharding.Coordinator
+
 	shouldProcessDataFunc func() bool
+
+	mutEquivalentMessages sync.RWMutex
+	msgCounters           map[string]uint64
 }
 
 // NewEquivalentMessagesDebugger returns a new instance of equivalentMessagesDebugger
-func NewEquivalentMessagesDebugger() *equivalentMessagesDebugger {
-	debugger := &equivalentMessagesDebugger{
+func NewEquivalentMessagesDebugger(proofsPool proofsPoolHandler, shardCoordinator sharding.Coordinator) (*equivalentMessagesDebugger, error) {
```

Comment:
this component will should track the equivalentProofsTracker somehow, perhaps in another PR

Reply:
yes, currently it tracks the proofs and keeps a counter for each messsage